.changeset/early-penguins-know.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": major
```

Comment:
I doubt if only type definition change should be considered as a breaking. Such a change does not break anything about runtime behavior but may raise a type-check error.

Do you have any (semver) guidelines about such a change?

Reply:
> Do you have any (semver) guidelines about such a change?

We don't. Is there a defacto approach for packages that publish types that we could adopt? 

If not, let's just switch back to refactor like you originally had it.

Reply:
> Is there a defacto approach for packages that publish types that we could adopt?

I'm not sure if it exists or not... 🤷🏼

Reply:
When I googled, the following website is found:

Semantic Versioning for TypeScript Types - https://www.semver-ts.org/

Reply:
When I read semver-ts.org, I feel a bit too strict, honestly.

However, aside from the future, because it's obvious that there's a possibility of breaking anything, I have no objections to considering this PR as breaking. 👍🏼